### DISCUSSION TITLE
Use pre-commit for running git hooks

### DISCUSSION THREAD

**Discussion by alex-oleshkevich** [encode/starlette#1900] [Ideas] [answered]
Currently we use shell scripts to run test and linters. There is an excellent (and very convenient) tool for this purpose called pre-commit - https://pre-commit.com/
The tool will run linters on git commit.

Besides of flake, black, and pytest it supports a large list of built-in linters.

Sample config - https://github.com/alex-oleshkevich/starsessions/blob/master/.pre-commit-config.yaml

Would you mind to integrate it?

**Comment by Kludex** [comment, marked as answer]
@tomchristie was against it, not sure if he has changed his mind...

I'm happy with the tooling we have in place. :man_shrugging:

**Reply by lovelydinosaur** [reply]
> not sure if he has changed his mind...

My mind remains unchanged.

**Reply by lovelydinosaur** [reply]
On this particular point, at least.
